### PR TITLE
Update Ecker2015 solid-phase diffusivity functions to more closely match the paper

### PR DESCRIPTION
**Pull request by DrSOKane**
The solid-phase diffusivity functions in `Ecker2015` and `Ecker2015_graphite_halfcell` are updated to better match the functions in the original paper. A new citation `Yuan2023`, for which these parameters were used, is also added.

Advantages:
* The new functions reproduce phase separation behaviour in a similar way to the Cahn-Hilliard equations
* With them, the results in `Yuan2023` can be reproduced.

Disadvantages:
* This level of physical accuracy comes at a noticeable computational cost, comparable with that of `ORegan2022`
* The results in `Richardson2019` are no longer reproduced. However, that paper still made a major contribution towards our understanding of the `Ecker2015` parameters and is therefore still cited.

### CODE REVIEW

**Comment by rtimms** [comment]
My preference would be to add a new `Yuan2023` parameter set so that the Ecker results don't change from version to version.

**Review by valentinsulzer — COMMENTED** [review]
What's the formula for D_s_n and D_s_p actually used by Ecker? I can't find it in Part I or Part II.

File: CHANGELOG.md [hunk 1]
```diff
@@ -47,6 +48,7 @@
 
 ## Breaking changes
 
+- Updated solid-phase diffusivity functions in `Ecker2015` and `Ecker2015_graphite_halfcell` parameter sets to better match the original papers ([#4141](https://github.com/pybamm-team/PyBaMM/pull/4141))
```
Comment (kratman) [inline CHANGELOG.md:51]: This should probably be under only "breaking changes" or "features" instead of both

**Comment by kratman** [comment]
@DrSOKane Is this one fixed by this change as well?

#2152

**Comment by DrSOKane** [comment]
> What's the formula for D_s_n and D_s_p actually used by Ecker? I can't find it in Part I or Part II.

They don't give a formula, only data. It's possible that they used an interpolation between the data points. LiionDB has this data (I put it there!) so an alternative approach would be to use that.

**Comment by kratman** [comment]
@DrSOKane This PR was marked as merged because the commits were also present in #4168

**Pull request merged**